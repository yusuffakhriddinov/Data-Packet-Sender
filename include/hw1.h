/*
ANY CHANGES MADE TO THIS FILE WILL BE DISCARDED DURING GRADING!
*/

#ifndef HW1_H
#define HW1_H

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void print_packet_sf(unsigned char packet[]);
unsigned int compute_checksum_sf(unsigned char packet[]);
unsigned int reconstruct_array_sf(unsigned char *packets[], unsigned int packets_len, int *array, unsigned int array_len);
unsigned int packetize_array_sf(int *array, unsigned int array_len, unsigned char *packets[], unsigned int packets_len,
                          unsigned int max_payload, unsigned int src_addr, unsigned int dest_addr, 
                          unsigned int src_port, unsigned int dest_port, unsigned int maximum_hop_count,
                          unsigned int compression_scheme, unsigned int traffic_class);

#endif // HW1_H
