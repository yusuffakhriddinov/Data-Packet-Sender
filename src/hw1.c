#include "hw1.h"

void print_packet_sf(unsigned char packet[])
{
    (void)packet;
}

unsigned int compute_checksum_sf(unsigned char packet[])
{
    (void)packet;
    return -1;
}

unsigned int reconstruct_array_sf(unsigned char *packets[], unsigned int packets_len, int *array, unsigned int array_len) {
    (void)packets;
    (void)packets_len;
    (void)array;
    (void)array_len;
    return -1;
}

unsigned int packetize_array_sf(int *array, unsigned int array_len, unsigned char *packets[], unsigned int packets_len,
                          unsigned int max_payload, unsigned int src_addr, unsigned int dest_addr,
                          unsigned int src_port, unsigned int dest_port, unsigned int maximum_hop_count,
                          unsigned int compression_scheme, unsigned int traffic_class)
{
    (void)array;
    (void)array_len;
    (void)packets;
    (void)packets_len;
    (void)max_payload;
    (void)src_addr;
    (void)dest_addr;
    (void)src_port;
    (void)dest_port;
    (void)maximum_hop_count;
    (void)compression_scheme;
    (void)traffic_class;
    return -1;
}
