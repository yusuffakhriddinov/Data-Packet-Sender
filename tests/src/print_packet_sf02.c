#include <stdio.h>

#include "hw1.h"

int main() {
    unsigned char packet[] = {
        0x0e,0x4e,0x1c,0x00,0xd5,0x9f,0x80,0x87,0x54,0xe0,0x02,0x49,0x81,0x2a,0x25,0x1a,0x00,0x01,0x62,0xbc,0x00,0x0c,0xa2,0x4e,0xff,0xff,0xdc,0x5d,0x00,0x00,0x1d,0xd2,0x00,0x06,0xc8,0x1c,
		};
    print_packet_sf(packet);
    return 0;
}
