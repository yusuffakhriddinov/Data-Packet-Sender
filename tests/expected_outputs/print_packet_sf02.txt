Source Address: 15000000
Destination Address: 14000000
Source Port: 8
Destination Port: 7
Fragment Offset: 5432
Packet Length: 36
Maximum Hop Count: 19
Checksum: 76325
Compression Scheme: 0
Traffic Class: 26
Payload: 90812 827982 -9123 7634 444444
